File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int n=nums.size();
        vector<bool> ans(n,false);
        int mod=0;

        for(int i=0;i<n;i++){
            mod=(mod*2+ nums[i])%5;
            if(mod==0) ans[i]=true;
        }

        return ans;
    }
};

File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.js
/**
 * @param {number[]} nums
 * @return {boolean[]}
 */
var prefixesDivBy5 = function(nums) {
    let ans=[];
    let n=nums.length;
    let mod=0;

    for(let i=0;i<n;i++){
        mod=(mod*2 + nums[i])%5;
        if(mod===0) ans.push(true);
        else{
            ans.push(false);
        }
    }

    return ans;
};